### PR TITLE
Add AddToPlaylistDrawer

### PR DESCRIPTION
**Pull request by Kyle-Shanks**
### Description

Added Add to Playlist drawer component

### How Has This Been Tested?

Manually tested. There is one small weirdness with scrolling the first time that I can't quite figure out, but everything else works

### CODE REVIEW

File: src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+      playlist_name: trackTitle,
+      is_private: false
+    })
+    const tempId = `${Date.now()}`
+    dispatchWeb(
+      createPlaylist(tempId, metadata, CreatePlaylistSource.FROM_TRACK, trackId)
+    )
+    dispatchWeb(addTrackToPlaylist(trackId!, tempId))
+    toast({ content: messages.createdToast })
+    // TODO: Update the fromPage location. Need to figure out how to get page that the user came from
```
Comment (Kyle-Shanks) [inline src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx:99]: Ah, forgot about this, will fix

File: src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+const messages = {
+  title: 'Add To Playlist',
+  addedToast: 'Added To Playlist!',
+  createdToast: 'Playlist Created!'
+}
+
+const createStyles = (themeColors: ThemeColors) =>
+  StyleSheet.create({
+    cardContainer: {
+      display: 'flex',
```
Comment (dylanjeffers) [inline src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx:45]: @sliptype was mentioning that `display: 'flex'` is a default for RN

File: src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+          startColor='rgba(133,129,153,0.16)'
+        >
+          <Button
+            title='Create New Playlist'
+            onPress={addToNewPlaylist}
+            type={ButtonType.COMMON}
+          />
+        </Shadow>
+        <View style={styles.cardContainer}>
+          {userPlaylists.map((playlist, idx) => (
```
Comment (dylanjeffers) [inline src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx:136]: we interested in doing a FlatList here, or were there scroll issues?

Comment (dylanjeffers) [reply src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx:136]: oh yeah nice, we have a cool helper now, i think you can use `CardList` component!

File: src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+import { useThemedStyles } from 'app/hooks/useThemedStyles'
+import { ThemeColors } from 'app/utils/theme'
+
+const messages = {
+  title: 'Add To Playlist',
+  addedToast: 'Added To Playlist!',
+  createdToast: 'Playlist Created!'
+}
+
+const createStyles = (themeColors: ThemeColors) =>
```
Comment (dylanjeffers) [inline src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx:42]: not a huge deal, but we have the `makeStyles` helper, which can comes with a spacing util and palette!

File: src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+            type={ButtonType.COMMON}
+          />
+        </Shadow>
+        <View style={styles.cardContainer}>
+          {userPlaylists.map((playlist, idx) => (
+            <Card
+              key={playlist.playlist_id}
+              style={[styles.card, { marginLeft: idx % 2 ? 8 : 0 }]}
+              id={playlist.playlist_id}
+              type='collection'
```
Comment (dylanjeffers) [inline src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx:141]: Can we use the `CollectionCard` component here by chance?

File: src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+      is_private: false
+    })
+    const tempId = `${Date.now()}`
+    dispatchWeb(
+      createPlaylist(tempId, metadata, CreatePlaylistSource.FROM_TRACK, trackId)
+    )
+    dispatchWeb(addTrackToPlaylist(trackId!, tempId))
+    toast({ content: messages.createdToast })
+    // TODO: Update the fromPage location. Need to figure out how to get page that the user came from
+    pushRouteWeb(playlistPage(user.handle, trackTitle, tempId), FEED_PAGE)
```
Comment (dylanjeffers) [inline src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx:100]: also a headsup! we have a new `useNavigation` hook that can handle the pushRouteWeb and the react-navigation useNavigate, which we could potentially use here, there is an example in the `TrackTile` component for `goToArtistPage`

**Review by dylanjeffers — APPROVED** [review]
lgtm though!

**Review by sliptype — APPROVED** [review]
Looks good! I agree with the things @dylanjeffers said

File: src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx (hunk 6 above)
Comment (Kyle-Shanks) [reply src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx:62]: Going to make this update in another PR. Need to update the routes in order to go directly to a user playlist page which is not set up just yet

**Pull request merged**